### PR TITLE
src/benchmarks: deprecate JEventProcessorSequential::PrefetchT in favor of PODIO collections

### PR DESCRIPTION
**Pull request by veprbl**
PrefetchT relies on compatibility interface that copies PODIO pointer objects into `std::vector<> JFactory::mData` member. We'd like for all of our code to work with PODIO collections directly.

### CODE REVIEW

File: src/benchmarks/reconstruction/lfhcal_studies/lfhcal_studiesProcessor.cc [hunk 1]
```diff
@@ -313,15 +312,15 @@ void lfhcal_studiesProcessor::Process(const std::shared_ptr<const JEvent>& event
...
+  const auto &simHits = *static_cast<const edm4hep::SimCalorimeterHitCollection*>(event->GetCollectionBase(nameSimHits));
+  for (const auto caloHit : simHits) {
+    float x         = caloHit.getPosition().x / 10.;
+    float y         = caloHit.getPosition().y / 10.;
+    float z         = caloHit.getPosition().z / 10.;
+    uint64_t cellID = caloHit.getCellID();
+    float energy    = caloHit.getEnergy();
     double time = std::numeric_limits<double>::max();
-    for (const auto& c : caloHit->getContributions()) {
+    for (const auto c : caloHit.getContributions()) {
```
Comment (wdconinc) [inline src/benchmarks/reconstruction/lfhcal_studies/lfhcal_studiesProcessor.cc:323]: Maybe -Wpedantic but should we keep the reference?
```suggestion
    for (const auto& c : caloHit.getContributions()) {
```
(applies in other places too)

Comment (veprbl) [reply src/benchmarks/reconstruction/lfhcal_studies/lfhcal_studiesProcessor.cc]: Good catch! Although, I only see one place where this needed to be fixed.

File: src/benchmarks/reconstruction/femc_studies/femc_studiesProcessor.cc [hunk 2]
```diff
@@ -482,15 +481,15 @@ void femc_studiesProcessor::Process(const std::shared_ptr<const JEvent>& event)
     m_log->info("-----> found {} clusters" , clusters_calo.size());
     hRecNClusters_E_eta->Fill(mcenergy, clusters_calo.size(), mceta);
     int iCl = 0;
-    for (auto& cluster : clusters_calo) {
+    for (const auto cluster : clusters_calo) {
```
Comment (wdconinc) [inline src/benchmarks/reconstruction/femc_studies/femc_studiesProcessor.cc:484]: Applies here.

File: src/benchmarks/reconstruction/femc_studies/femc_studiesProcessor.cc [hunk 3]
```diff
@@ -482,15 +481,15 @@ void femc_studiesProcessor::Process(const std::shared_ptr<const JEvent>& event)
...
       if (iCl < maxNCluster && enableTreeCluster){
         t_fEMC_cluster_E[iCl]       = (float)cluster.cluster_E;
         t_fEMC_cluster_NCells[iCl]  = (int)cluster.cluster_NTowers;
         t_fEMC_cluster_Eta[iCl]     = (float)cluster.cluster_Eta;
         t_fEMC_cluster_Phi[iCl]     = (float)cluster.cluster_Phi;
       }
       hRecClusterEcalib_E_eta->Fill(mcenergy, cluster.cluster_E/mcenergy, mceta);
-      for (auto & cluster_tower : cluster.cluster_towers){
+      for (const auto cluster_tower : cluster.cluster_towers){
```
Comment (wdconinc) [inline src/benchmarks/reconstruction/femc_studies/femc_studiesProcessor.cc:492]: Here.

File: src/benchmarks/reconstruction/lfhcal_studies/lfhcal_studiesProcessor.cc [hunk 4]
```diff
@@ -625,10 +624,10 @@ void lfhcal_studiesProcessor::Process(const std::shared_ptr<const JEvent>& event
   hRecFNClusters_E_eta->Fill(mcenergy, iClF, mceta);
   // fill hists for highest Island cluster
   iClF          = 0;
-  for (auto& cluster : lfhcalClustersF) {
+  for (const auto cluster : lfhcalClustersF) {
```
Comment (wdconinc) [inline src/benchmarks/reconstruction/lfhcal_studies/lfhcal_studiesProcessor.cc:627]: Here.

File: src/benchmarks/reconstruction/lfhcal_studies/lfhcal_studiesProcessor.cc [hunk 5]
```diff
@@ -643,31 +642,31 @@ void lfhcal_studiesProcessor::Process(const std::shared_ptr<const JEvent>& event
 
   if (enableECalCluster){
     try {
-      auto fEMCClustersF = event->Get<edm4eic::Cluster>("EcalEndcapPClusters");
+      const auto &fEMCClustersF = *static_cast<const edm4eic::ClusterCollection*>(event->GetCollectionBase("EcalEndcapPClusters"));
       m_log->info("-----> found fEMCClustersF:" , fEMCClustersF.size());
-      for (auto& cluster : fEMCClustersF) {
+      for (const auto cluster : fEMCClustersF) {
```
Comment (wdconinc) [inline src/benchmarks/reconstruction/lfhcal_studies/lfhcal_studiesProcessor.cc:647]: Here.

**Review by wdconinc — COMMENTED** [review]
Marked some more. As I said, it's pedantic and not performance critical of course.

**Comment by wdconinc** [comment]
(Really, I just want to be able to say I have reviewed code from the Canadian seat of government, Parliament Hill.)

File: src/benchmarks/reconstruction/femc_studies/femc_studiesProcessor.cc (hunk 2 above)
Comment (veprbl) [reply src/benchmarks/reconstruction/femc_studies/femc_studiesProcessor.cc:484]: But the PODIO collection iterators don't return a reference https://github.com/AIDASoft/podio/blob/babec745c12727cd5da590abbe1e3cf1ab8743ad/python/templates/macros/iterator.jinja2#L33-L36

Comment (wdconinc) [reply src/benchmarks/reconstruction/femc_studies/femc_studiesProcessor.cc:484]: Right. I keep thinking they are more std::vector than they actually are.

**Review by wdconinc — APPROVED** [review]

**Pull request merged**